### PR TITLE
Fix view roles command to prevent it showing excess roles

### PR DESCRIPTION
**Pull request by mini-ninja-64**
* Refactor roles.js for readability
* Fix a problem with viewRoles showing unnasignable roles
* Update role assertion to be done on id where possible
* Let users remove roles even if they are blacklisted (excluding managed roles)

### CODE REVIEW

File: commands/roles.js [hunk 1]
```diff
@@ -30,23 +70,25 @@ const addRole = async (interaction, guildSettings) => {
...
-			(role) => role.name === chosenRole.name
-		)
-	) {
+	if (!isManagedRole(roleProvided)) {
+		return interaction.editReply({
+			embeds: [embed("💔 **You can't remove that role.**")],
+		});
+	}
+
+	if (!memberHasRole(interaction.member, roleProvided)) {
```
Comment (amberstarlight) [inline commands/roles.js:81]: This should be above L75 in my opinion; we should check if the user has the role before we check if we can remove it.

File: commands/roles.js [hunk 2]
```diff
@@ -30,23 +70,25 @@ const addRole = async (interaction, guildSettings) => {
...
 const removeRole = async (interaction, guildSettings) => {
-	const chosenRole = getChosenRole(interaction, guildSettings);
+	const roleProvided = interaction.options.getRole('role');
 
-	if (
-		!interaction.member.roles.cache.some(
-			(role) => role.name === chosenRole.name
-		)
-	) {
+	if (!isManagedRole(roleProvided)) {
```
Comment (amberstarlight) [inline commands/roles.js]: Should this be the inverse? If it's not a managed role we should go ahead and remove it.

File: commands/roles.js (hunk 1 above)
Comment (amberstarlight) [inline commands/roles.js]: This should be above L75 in my opinion; we should check if the user has the role before we check if we can remove it.

**Review by amberstarlight — CHANGES_REQUESTED** [review]

File: commands/roles.js (hunk 2 above)
Comment (mini-ninja-64) [reply commands/roles.js]: Yes it should, thank you I have fixed this now

File: commands/roles.js (hunk 1 above)
Comment (mini-ninja-64) [reply commands/roles.js]: I agree it is more intuitive for users

**Review by amberstarlight — APPROVED** [review]
Great refactoring! I've tested this using the dev bot and I'm approving under the following conditions:

- We spoke about the UX and that it was out of scope for this fix, and will be addressed in a future PR, and
- I have tested all the user journeys in this command and the flows have all worked without error.

Let's get this merged. 🚀

**Pull request merged**